src/modules/apps/builder/ArtifactSharedIframe.tsx

```diff
@@ -51,20 +53,55 @@ export function ArtifactSharedIframe({ sourceCode }: Props) {
     postMessage({ type: PostMessageType.UPDATE_CODE, code });
   }, []);
 
-  const handleMessage = useCallback((event: MessageEvent<StliteMessage>) => {
-    const { origin, data } = event;
-
-    if (origin !== removeTrailingSlash(USERCONTENT_SITE_URL)) {
-      return;
-    }
-
-    if (
-      data.type === SCRIPT_RUN_STATE_CHANGED &&
-      data.scriptRunState === ScriptRunState.RUNNING
-    ) {
-      setState(State.READY);
-    }
-  }, []);
+  const handleMessage = useCallback(
+    async (event: MessageEvent<StliteMessage>) => {
+      const { origin, data } = event;
+
+      if (origin !== removeTrailingSlash(USERCONTENT_SITE_URL)) {
+        return;
+      }
+
+      if (
+        data.type === RecieveMessageType.SCRIPT_RUN_STATE_CHANGED &&
+        data.scriptRunState === ScriptRunState.RUNNING
+      ) {
+        setState(State.READY);
+      }
+      if (data.type === RecieveMessageType.REQUEST) {
+        switch (data.request_type) {
+          case '/modules-to-packages':
```

Comment:
We should probably pick normal names for the requests that don't look like URLs since they won't correspond to endpoint calls 1:1 either way